go-controller/pkg/libovsdb/libovsdb.go

```diff
@@ -91,10 +98,10 @@ func NewNBClientWithConfig(cfg config.OvnAuthConfig, stopCh <-chan struct{}) (cl
 	if err != nil {
 		return nil, err
 	}
-	return newClientWithMonitor(cfg, stopCh, dbModel)
+	return newClientWithMonitor(cfg, stopCh, dbModel, nil)
 }
 
-func newClientWithMonitor(cfg config.OvnAuthConfig, stopCh <-chan struct{}, dbModel *model.ClientDBModel) (client.Client, error) {
+func newClientWithMonitor(cfg config.OvnAuthConfig, stopCh <-chan struct{}, dbModel *model.ClientDBModel, tables []model.Model) (client.Client, error) {
```

Comment:
nit: IMHO this function generalises the client setup too much and makes the code harder to follow. this isn't a direct criticim of this commit as it was the case beforehand too, although it was less egregious since the codepath was identical

since there are just 2 ways of setting up a client (NB and SB) and it's very unlikely we'll ever have a third type, in https://github.com/ovn-org/ovn-kubernetes/commit/79fba189c11fedb811b0741c32a7490a158871b3 I opted to inline what this function was doing at the end of the `NewNBClientWithConfig` and `NewSBClientWithConfig` functions. The duplicated code is minimal and  the benefit is that, to my eyes at least, it's easier to read since the information you need (which tables are being monitored) is right there, not indirected via a generalised setup function and hidden in a  package-level variable.

this is just my opinion :wink: 
like i said before, I wasn't aware of this when I threw a commit on my patch to fix this.
the priority is to not monitor all of sbdb, and the "how" is up to the people that have to maintain this

Reply:
OK, I agree that your approach might be better. I am closing this PR in preference of yours